lib/ansible/vars/reserved.py

```diff
@@ -1,73 +1,66 @@
-# (c) 2017 Ansible By Red Hat
-#
-# This file is part of Ansible
-#
-# Ansible is free software: you can redistribute it and/or modify
-# it under the terms of the GNU General Public License as published by
-# the Free Software Foundation, either version 3 of the License, or
-# (at your option) any later version.
-#
-# Ansible is distributed in the hope that it will be useful,
-# but WITHOUT ANY WARRANTY; without even the implied warranty of
-# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-# GNU General Public License for more details.
-#
-# You should have received a copy of the GNU General Public License
-# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+# coding: utf-8
+# Copyright: (c) 2017, Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
 
 # Make coding more python3-ish
 from __future__ import (absolute_import, division, print_function)
 __metaclass__ = type
 
+from ansible import constants as C
+from ansible.errors import AnsibleError
+from ansible.module_utils._text import to_text
 from ansible.playbook import Play
 from ansible.playbook.block import Block
 from ansible.playbook.role import Role
 from ansible.playbook.task import Task
+from ansible.template import Templar
 from ansible.utils.display import Display
 
 display = Display()
 
+_INTERNAL_HARDCODED = frozenset(('local_action',))
+_e = Templar(None).environment
+_JINJA_RESERVED = frozenset(set(_e.globals).union(_e.filters, _e.tests))
+# FIXME: remove these exceptions if we can
+_RESERVE_EXCEPTIONS = frozenset(('environment', 'gather_subset', 'vars'))
+# FIXME: with_<lookups>
```

Comment:
could be combo of `with_` X `_e.lookups`

Reply:
I guess I could do `[l.__module__.split('.')[-1] for l in lookup_loader.all()]` to get the list of lookup plugins names.

Reply:
Done.